src/common.rs

```diff
@@ -981,9 +982,87 @@ pub fn get_current_pid() -> Result<Pid, &'static str> {
     inner()
 }
 
+/// MAC address for network interface
+#[derive(PartialEq, Eq, Copy, Clone)]
+pub struct MacAddr {
+    data: [u8; 6],
+}
+
+impl MacAddr {
+    pub const UNSPECIFIED: Self = MacAddr::new();
+
+    #[inline]
+    pub(crate) const fn new() -> Self {
+        Self { data: [0; 6] }
+    }
+
+    pub fn data(&self) -> &[u8; 6] {
+        &self.data
+    }
+}
+
+impl From<[u8; 6]> for MacAddr {
+    fn from(data: [u8; 6]) -> Self {
+        Self { data }
+    }
+}
+
+impl FromStr for MacAddr {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut data = [0; 6];
+        for (index, result) in s
+            .split(':')
+            .take(6)
+            .map(|x| u8::from_str_radix(x, 16))
+            .enumerate()
+        {
+            match result {
+                Ok(result) => {
+                    data[index] = result;
+                }
+                Err(_) => {
+                    break;
+                }
+            }
+            if index == 5 {
+                return Ok(MacAddr { data });
+            }
+        }
+
+        Err("invalid MAC address syntax")
+    }
+}
+
+impl std::fmt::Display for MacAddr {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let s = self
+            .data
+            .iter()
+            .map(|b| format!("{:02x}", *b))
+            .collect::<Vec<String>>()
+            .join(":");
+        f.write_str(&s)
+    }
+}
```

Comment:
This implementation is quite slow because it does a lot of allocations. Instead you could do:

```suggestion
impl fmt::Display for MacAddr {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(
            f,
            "{:02x}:{:02x}:{:02x}:{:02x}:{:02x}:{:02x}",
            self.data[0],
            self.data[1],
            self.data[2],
            self.data[3],
            self.data[4],
            self.data[5],
        )
    }
}
```